File: programmers/고득점Kit/정렬/H-index.js
// H-index
// https://school.programmers.co.kr/learn/courses/30/lessons/42747

function solution(citations) {
  let answer = 0;
  for (let h = 1; h < Math.max(...citations); h++) {
    let count = 0;
    for (let j = 0; j < citations.length; j++) {
      if (h <= citations[j]) count += 1;
    }
    if (h <= count) answer = h;
    else return answer;
  }

  return answer;
}

console.log(solution([3, 0, 6, 1, 5]));
console.log(solution([3, 4]));
console.log(solution([4, 3, 3, 3, 0, 0, 0]));

File: programmers/Lv.2/H-Index.py
# H-Index
# https://school.programmers.co.kr/learn/courses/30/lessons/42747

def solution(citations):
    citations.sort(reverse=True)
    for i in range(citations[0], -1, -1):
        count = 0
        for j in range(len(citations)):
            if i <= citations[j]:
                count += 1
            else:
                break
        if count >= i:
            return i

print(solution([3, 0, 6, 1, 5]))
print(solution([0]))